feedback_prompt_learning/data/task/base_task.py

```
@@ -0,0 +1,333 @@
+        data_type: str | set
+            str: preds, labels are List(str)
+            set: preds, labels are List(set)
+            
+        Every time a batch data is sampled and predicted, by comparing with
+        the labels, PromptAgent collect the errors.
+        Function called at: prompt_optim_agent/world_model/gradient_descent.py line 54
+        
+        '''
+        if data_type == "set":
+            comparisons = []
+            for p, l in zip(preds, labels):
```

Ruff (0.14.7)

Ambiguous variable name: `l`

(E741)